mem/mem.go

```diff
@@ -26,41 +26,61 @@ import (
 	"strconv"
 	"strings"
 	"time"
+	"unicode"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/mitchellh/mapstructure"
 
 	"github.com/intelsdi-x/snap/control/plugin"
 	"github.com/intelsdi-x/snap/control/plugin/cpolicy"
 	"github.com/intelsdi-x/snap/core"
+
+	"github.com/intelsdi-x/snap-plugin-utilities/config"
+	"github.com/intelsdi-x/snap-plugin-utilities/ns"
 )
 
 const (
 	// VENDOR namespace part
-	VENDOR = "intel"
+	pluginVendor = "intel"
 	// CLASS namespace part
-	CLASS = "procfs"
+	fs = "procfs"
 	// PLUGIN name namespace part
-	PLUGIN = "meminfo"
+	pluginName = "meminfo"
 	// VERSION of mem info plugin
-	VERSION = 2
+	pluginVersion = 3
 )
 
-var memInfo = "/proc/meminfo"
-
 // New creates instance of mem info plugin
 func New() *memPlugin {
-	mp := &memPlugin{stats: map[string]interface{}{}}
-	return mp
+	logger := log.New()
+	return &memPlugin{logger: logger}
+}
+
+// Meta returns plugin meta data
+func Meta() *plugin.PluginMeta {
+	return plugin.NewPluginMeta(
+		pluginName,
+		pluginVersion,
+		plugin.CollectorPluginType,
+		[]string{},
+		[]string{plugin.SnapGOBContentType},
+		plugin.ConcurrencyCount(1),
+	)
 }
 
 // GetMetricTypes returns list of available metric types
 // It returns error in case retrieval was not successful
-func (mp *memPlugin) GetMetricTypes(_ plugin.ConfigType) ([]plugin.MetricType, error) {
+func (mp *memPlugin) GetMetricTypes(cfg plugin.ConfigType) ([]plugin.MetricType, error) {
```

Comment:
Don't we want to avoid failing while loading plugins? In this implementation plugin builds list of metrics in ```GetMetricTypes``` statically, so it does not expect any configuration to be prepared in global config. Potential configuration can come with task manifest.